### ISSUE TITLE
VSCode Not Connecting To DevTools - Widget Inspector: Not Found - [VSCode]

### ISSUE THREAD

**Issue by Noahbanderson** [flutter/devtools#4417]
When running any of my flutter apps and trying to debug them, I am presented with a blank panel with the text: "Not Found".

This first began when I updated flutter from v2.x to v3.x, however, even when trying to revert to a previous version I still get this same panel.

![Screen Shot 2022-08-26 at 5 38 30 PM](https://user-images.githubusercontent.com/52230969/186999402-50096098-0b68-4d6c-9708-dedfe6e2f0b8.png)

I've inspected the WebView to see what information I can gather from it.

https://user-images.githubusercontent.com/52230969/187007321-f968d6dd-e605-414f-9514-58c1b5c6cb66.mov

I've taken the URL `http://127.0.0.1:9100/?uri=...&page=inpsector...` and put it in my browser only to get the same "Not Found" result. In the Network tab, in the Initiator section, it looks like the link is being altered. And indeed it is, see [here](https://github.com/flutter/devtools/blob/master/packages/devtools_app/lib/src/primitives/url_utils.dart) in the `mapLegacyUrl` function. I place that altered URL in the browser and it gives the same "Not Found" message.

![network_initiator_section](https://user-images.githubusercontent.com/52230969/187008150-e58617a4-990b-4993-a1e1-7b857c4800b6.png)

If I change the url to the format `http://127.0.0.1:9100/#/?uri=...&page=inpsector...` (emphasis on the '#'), the browser is able to connect to the devtools server without issue. The only thing I can conclude from this is that devtools is able to serve older paths but not newer ones, even though it is insistent to translate an old path to a newer path.

Things I have done to fix the issue:
1) Removed Flutter and Dart packages from VSCode. Looked through my VSCode settings to see if any weird settings was altered.
2) Uninstalled FlutterSDK and removed .pub-cache folder
3) Restarted computer
4) Reinstalled FlutterSDK and VSCode extensions

Problem still persisted.

I am unable to replicate this issue on another machine.

`flutter doctor -v` output

```
Development » flutter doctor -v
[✓] Flutter (Channel stable, 3.0.5, on macOS 12.5.1 21G83 darwin-x64, locale en-US)
    • Flutter version 3.0.5 at /Users/noahanderson/Development/tools/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f1875d570e (6 weeks ago), 2022-07-13 11:24:16 -0700
    • Engine revision e85ea0e79c
    • Dart version 2.17.6
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/noahanderson/Library/Android/sdk
    • Platform android-33, build-tools 30.0.3
    • ANDROID_HOME = /Users/noahanderson/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] VS Code (version 1.70.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.46.0

[✓] Connected device (4 available)
    • SM G781U (mobile)      • RFCN90GHDYV                          • android-arm64  • Android 11 (API 30)
    • iPhone 12 Pro (mobile) • 971CD230-BC5F-484E-AD1C-27A443D25195 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-15-5 (simulator)
    • macOS (desktop)        • macos                                • darwin-x64     • macOS 12.5.1 21G83 darwin-x64
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 104.0.5112.101

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

**Comment by kenzieschmoll** [comment]
@DanTup

**Comment by DanTup** [comment]
I think there's some caching issue here, though I'm unable to reproduce. It came up in Discord (which led to https://github.com/Dart-Code/Dart-Code/issues/4008 being filed).

As a workaround, you can force DevTools onto another port by setting the "Dev Tools Port" setting in VS Code to something not in the range 9100-9110:

<img width="771" alt="Screenshot 2022-09-01 at 10 16 37" src="https://user-images.githubusercontent.com/1078012/187878713-9995318e-d471-4a68-95f1-a3ef2c001e7d.png">

This will prevent any previously cached version of DevTools being used. It's possible this is the same as https://github.com/flutter/devtools/issues/3896 and once fixed this will be fixed too, but I'm not certain.

**Comment by Noahbanderson** [comment]
Thank you so much for support. Changing the port number worked like a charm.

A new stable flutter version was just released, and with it came DevTools 2.15.0. I removed the port number from the VSCode settings and it still worked like normal. Just an FYI.

**Comment by DanTup** [comment]
That's good to know, thanks!
